src/main/java/io/github/cottonmc/resources/CottonResources.java

```diff
@@ -36,239 +52,192 @@
 		if (file.exists()) {
 			CONFIG = loadConfig();
 			saveConfig(CONFIG);
 		} else {
 			saveConfig(CONFIG);
 		}
 	}
-	
+
 	private static void setupBiomeGenerators() {
 		for (Biome biome : Registry.BIOME) {
 			setupBiomeGenerator(biome);
 		}
 	}
-	
+
 	private static void setupBiomeGenerator(Biome biome) {
 		biome.addFeature(GenerationStep.Feature.UNDERGROUND_ORES,
-			Biome.configureFeature(
-				CottonOreFeature.COTTON_ORE,
-				FeatureConfig.DEFAULT,
-				Decorator.COUNT_RANGE,
-				new RangeDecoratorConfig(1, 0, 0, 256)
-			)
-		);
-	}
-
-	private static MetalResourceType builtinMetal(String id, Supplier<Block> oreSupplier, String... extraAffixes) {
-		MetalResourceType result = new MetalResourceType(id);
-		if (oreSupplier != null) {
-			result.withOreSupplier(oreSupplier);
-			result.withBlockAffix("nether_ore", oreSupplier);
-			result.withBlockAffix("end_ore", oreSupplier);
-		}
-
-		if (extraAffixes.length > 0) {
-			result.withItemAffixes(extraAffixes);
-		}
-
-		BUILTINS.put(id, result);
-		return result;
-	}
-
-	private static void builtinGem(String id, Supplier<Block> oreSupplier, String... extraAffixes) {
-		GemResourceType result = new GemResourceType(id).withOreSupplier(oreSupplier);
-		if (extraAffixes.length > 0) {
-			result.withItemAffixes(extraAffixes);
-		}
-
-		BUILTINS.put(id, result);
+			CottonOreFeature.COTTON_ORE
+				.configure(FeatureConfig.DEFAULT)
+				.createDecoratedFeature(
+					Decorator.COUNT_RANGE.configure(new RangeDecoratorConfig(1, 0, 0, 256)
+					)
+				));
 	}
 
-	private static RadioactiveResourceType builtinRadioactive(String id, Supplier<Block> oreSupplier, String... extraAffixes) {
-		RadioactiveResourceType result = new RadioactiveResourceType(id);
-		if (oreSupplier != null) result.withOreSupplier(oreSupplier);
-
-		if (extraAffixes.length > 0) {
-			result.withItemAffixes(extraAffixes);
-		}
-
-		BUILTINS.put(id, result);
-		return result;
-	}
-
-	private static void builtinItem(String id, String... extraAffixes) {
-		GenericResourceType result = new GenericResourceType(id).withItemAffixes(extraAffixes);
-		if (extraAffixes.length == 0) {
-			result.withItemAffixes(""); //This is just a base type with no affixes, like "mercury".
-		}
-		BUILTINS.put(id, result);
-	}
-	
-	
 	public static CottonResourcesConfig loadConfig() {
-		File file = new File(FabricLoader.getInstance().getConfigDirectory(),"CottonResources.json5");
-		
+		File file = new File(FabricLoader.getInstance().getConfigDirectory(), "CottonResources.json5");
+
 		Jankson jankson = JanksonFactory.builder()
-				.registerTypeAdapter(OreGenerationSettings.class, OreGenerationSettings::deserialize)
-				.build();
+			.registerTypeAdapter(OreGenerationSettings.class, OreGenerationSettings::deserialize)
+			.build();
 		try {
 			JsonObject json = jankson.load(file);
-			System.out.println("Loading: "+json);
+			System.out.println("Loading: " + json);
```

Comment:
Should we use a proper logger, so the console actually shows that it's Cotton Resources doing this?

Reply:
Yes, I'll go over in a few mins and replace all the system.out references